src/lib/NumberIncrementer/index.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { FunctionComponent, useState, useCallback } from 'react'
+import { createCounterSubtractFactory, createCounterAddFactory } from './factory.counter'
+import clsx from 'clsx'
+
+import styles from './numberIncrementer.module.scss'
+
+interface Props {
+  className?: string
+  [rest: string]: unknown // ...rest property
+}
+
+const NumberIncrementer: FunctionComponent<Props> = ({ className, ...rest }) => {
+  const [number, setNumber] = useState(1)
+
+  const subtractNumber = useCallback(() => {
+    setNumber(createCounterSubtractFactory(number))
```

Comment:
Need to have an `onChange` prop on both this function and the addNumber function so that parent components can save the value

Reply:
Can you clarify a little further on this? I'm a bit confused on what I'm passing onChange to. Am I putting the onChange function in the callback or am I putting it on the actual button?